src/parser/mod.rs

```diff
@@ -2955,9 +2955,15 @@ impl<'a> Parser<'a> {
         }
     }
 
-    pub fn parse_create_view(&mut self, or_replace: bool) -> Result<Statement, ParserError> {
+    pub fn parse_create_view(
+        &mut self,
+        or_replace: bool,
+        temporary: bool,
+    ) -> Result<Statement, ParserError> {
         let materialized = self.parse_keyword(Keyword::MATERIALIZED);
         self.expect_keyword(Keyword::VIEW)?;
+        let if_not_exists = dialect_of!(self is SQLiteDialect|GenericDialect)
```

Comment:
Is this on purpose ? Most other constructs do not generate parse errors when parsed with the "wrong" dialect. Is there a rule for how sqlparser handles that, or does every contributor do it their own way ? 

 I was under the impression that dialects allowed disambiguating syntax that would parse differently in different databases, but did not cause a parse error when parsing unambiguous syntax from one dialect with a different one.

@alamb

Reply:
https://github.com/gabivlj/sqlparser-rs/blob/96c5979481536c6f6fe3193674059b2fff1d81b7/src/parser/mod.rs#L2984-L2990

For context, I took the idea from already existing RedshiftSqlDialect `with_no_schema_binding`. The intent here was excluding Dialects that do not support `CREATE VIEW IF NOT EXISTS`, for example the most similar feature for Postgresql is `CREATE OR REPLACE VIEW ` as `IF NOT EXISTS` on a view is not supported there.

Reply:
Yes, I see that. There does not seem to be a coherent rule over the entire codebase for whether database-specific syntax should cause a syntax error when parsed with a different dialect.

Reply:
> Yes, I see that. There does not seem to be a coherent rule over the entire codebase for whether database-specific syntax should cause a syntax error when parsed with a different dialect.

I agree this is not uniform in the codebase and  I think sqlparser-rs is more permissive than most other parsers in the sense that it will parse SQ that would generate a syntax error in the original implementation. This is broadly consistent with the sentiment of https://github.com/sqlparser-rs/sqlparser-rs#syntax-vs-semantics

In general I expect that the `GenericDialect` will parse the union of all other dialects, as long as they aren't in conflict with each other (e.g. the rules on how to parse strings vs identifiers differ in some dialects)